### ISSUE TITLE
Handle the remove/update share permissions response.

### ISSUE THREAD

**Issue by 2403905** [owncloud/web#10507]
## Describe the bug
Regarding the backend changes [https://github.com/owncloud/ocis/issues/8273](#8273) we should handle the remove/update share permissions response 423 when the file is locked.

## Steps to reproduce
1. admin created the `file.txt` and shared to einstein with the editor permissions.
2. einstein locked the file
```
curl -XLOCK 'https://localhost:9200/remote.php/dav/spaces/storage-users-1$some-admin-user-id-0000-000000000000!some-admin-user-id-0000-000000000000/file.txt' -ueinstein:relativity \
-H "Content-Type: application/xml" -H "timeout: Second-300" -d "<?xml version='1.0' encoding='UTF-8'?><d:lockinfo xmlns:d='DAV:'><d:lockscope><d:shared/></d:lockscope></d:lockinfo>" -vk
```
3. admin tries to remove/update share permissions.

## Expected behavior
admin get the message that the file is locked.

## Actual behavior
On the delete share action user recursively gets the removing dialog window

## Setup
local

<details>
<p>

```console
OCIS_XXX=somevalue
OCIS_YYY=somevalue
PROXY_XXX=somevalue
```

</p>
</details>

**Comment by ScharfViktor** [comment]
we disable on web delete rename or move action if file is locked
<img width="473" alt="Screenshot 2024-02-23 at 09 47 17" src="https://github.com/owncloud/web/assets/84779829/d5c46a72-6264-4631-accb-78f8c83ce2cf">

If possible, I would prefer that the user not have the ability to change or delete the resource share.

my expected behavior: 
<img width="478" alt="Screenshot 2024-02-23 at 09 48 54" src="https://github.com/owncloud/web/assets/84779829/64e540df-fe9a-4472-8fea-d7be7ff732cb">

**Comment by kulmann** [comment]
@AlexAndBear yes, that would be fine. The file name should have the lock information, both in the file list and in the right sidebar header section. As long as that shows the lock information (i.e. small icon with a lock) I wouldn't clutter the UI with even more lock information. Disabling `write` actions is sufficient then. 👍

**Comment by ScharfViktor** [comment]
re-tested. can confirm that case with file is fixed 
<img width="910" alt="image" src="https://github.com/owncloud/web/assets/84779829/8282a270-b8bd-48cc-9dde-a4b94851cd0a">

@kulmann @micbar I'm sorry to say, but we didn't have all the cases covered.
https://github.com/owncloud/ocis/issues/8273#issuecomment-1951979302 this workaround fixes only issue with file
IMHO we cannot disable change member role or change share role of folder if one file in the space or folder is locked

case 1 (project space)
steps: 
- `admin` created the `file.txt` and shared to `einstein` with the editor permissions.
- `einstein` locked the file
```
curl -XLOCK 'https://localhost:9200/remote.php/dav/spaces/projectSpaceUUID/file.txt' -ueinstein:relativity \
-H "Content-Type: application/xml" -H "timeout: Second-300" -d "<?xml version='1.0' encoding='UTF-8'?><d:lockinfo xmlns:d='DAV:'><d:lockscope><d:shared/></d:lockscope></d:lockinfo>" -vk
```
- `admin` removes/updates `einstein` space role.


case 2 (folder with files)
- `admin` created the `f1/file.txt` and shared folder`f1` to `einstein` with the editor permissions.
- `einstein` locked the file
```
curl -XLOCK 'https://localhost:9200/remote.php/dav/spaces/projectSpaceUUID/file.txt' -ueinstein:relativity \
-H "Content-Type: application/xml" -H "timeout: Second-300" -d "<?xml version='1.0' encoding='UTF-8'?><d:lockinfo xmlns:d='DAV:'><d:lockscope><d:shared/></d:lockscope></d:lockinfo>" -vk
```
- `admin` remove/update share permissions of folder `f1`


Actual: file is locked. Admin and another users cannot open this file:
<img width="1147" alt="image" src="https://github.com/owncloud/web/assets/84779829/ad7f4321-2033-4012-857d-76c01ab098ab">
